Makefile.inc

```diff
@@ -14,11 +14,22 @@ IDIRS	:=	${IDIRS}
 # Default is no man pages
 MAN	=
 
+# The SUBDIR_DEPTH variable (defined above) initializes tracking the
+# subdirectory depth, but it's not fully set at this stage in the build
+# process.  Instead, we jump back to the shell and print the (finalized)
+# value of SUBDIR_DEPTH.
+FINALIZED_SUBDIR_DEPTH	!=	${MAKE} -v SUBDIR_DEPTH
+
 .if !defined(NOLIBALL)
+.if exists(${FINALIZED_SUBDIR_DEPTH}/liball/optional_mutex_normal)
```

Comment:
I have no firm feelings about what we should have here.  We could spell out the full path to `liball_optional_mutex_normal.a` (or even that and `liball_optional_mutex_pthread.a`!).

I initially had the full path, but it looked messy, and I figured the directory would be sufficient to be confident that the project either had optional_mutex or not.